File: NutriSmart/app/src/main/java/com/example/nutrismart/data/model/UserDto.kt
package com.example.nutrismart.data.model

data class UserDto(
    val id: Long,
    val username: String,
    val email: String,
    val weight: Double? = null,
    val height: Double? = null,
    val dateOfBirth: String? = null,
    val targetWeight: Double? = null,
    val gender: Gender? = null,
    val activityLevel: ActivityLevel? = null,
    val maxDailyBudget: Double? = null,
    val dietaryPreferences: List<DietaryPreference>? = null,
    val medicalConditions: List<MedicalCondition>? = null,
    val dislikedFoods: List<String>? = null,
    val stepGoal: Int? = 10000,
    val isGoogleUser: Boolean = false,
    val isImperial: Boolean = false,
    val currency: Currency? = null
)

File: NutriSmartIOS/data/model/UserDto.swift
//
//  UserDto.swift
//  NutriSmartIOS
//

import Foundation

struct UserDto: Codable {
    let id: Int
    let username: String
    let email: String
    let weight: Double?
    let height: Double?
    let dateOfBirth: String?
    let targetWeight: Double?
    let gender: Gender?
    let activityLevel: ActivityLevel?
    let maxDailyBudget: Double?
    let dietaryPreferences: [DietaryPreference]?
    let medicalConditions: [MedicalCondition]?
    let dislikedFoods: [String]?
    let stepGoal: Int?
    let isGoogleUser: Bool
    let isImperial: Bool
    let currency: Currency?
}

struct UpdateUserRequest: Codable {
    let username: String?
    let weight: Double?
    let height: Double?
    let targetWeight: Double?
    let activityLevel: ActivityLevel?
    let maxDailyBudget: Double?
    let dietaryPreferences: [DietaryPreference]?
    let medicalConditions: [MedicalCondition]?
    let dislikedFoods: [String]?
    let stepGoal: Int?
    let isImperial: Bool?
    let currency: Currency?
}
